share/admin.py

```diff
@@ -353,8 +355,58 @@ class ShareUserAdmin(admin.ModelAdmin):
     inlines = (SourceAdminInline,)
 
 
+class SourceAddForm(forms.ModelForm):
+
+    class Meta:
+        model = Source
+        exclude = ('user', 'long_title', )
+        widgets = {'name': TextInput}
+
+    def validate_unique(self):
+        # Forces validation checks on every field
+        try:
+            self.instance.validate_unique()
+        except forms.ValidationError as e:
+            self._update_errors(e)
+
+    def _post_clean(self):
+        self.cleaned_data['long_title'] = self.cleaned_data.pop('name')[0]
+        self.cleaned_data['name'] = '.'.join(reversed(self.cleaned_data['home_page'].split('//')[1].split('.')))
```

Comment:
This will allow for `www` at the end of source `long_title`'s which isn't a blocker just slightly annoying.